app/_src/gateway/plugin-development/wasm.md

```diff
@@ -0,0 +1,113 @@
+---
+title: WebAssembly
+book: plugin_dev
+chapter: 11
+---
+
+# Your First Proxy-Wasm Filter
+
+Filters can be written in almost any language that supports WebAssembly as a
+compilation target, but the easiest and most ergonomic way to get started is by
+using a language which has an existing [Proxy-Wasm](https://github.com/proxy-wasm/spec)
+SDK, such as [Go](https://github.com/tetratelabs/proxy-wasm-go-sdk/)
+or [Rust](https://github.com/proxy-wasm/proxy-wasm-rust-sdk/).
+
+## Coding, building and deploying a Proxy-Wasm filter
+
+Kong made available Go and Rust Proxy-Wasm filter templates, which can be used
+to start coding your own filters or build as they are and have your first "hello
+world"-type filter:
+
+* [Go Proxy-Wasm filter template](https://github.com/Kong/proxy-wasm-go-filter-template/)
+* [Rust Proxy-Wasm filter template](https://github.com/Kong/proxy-wasm-rust-filter-template/)
+
+### Deploying the Filter with Kong
+
+After writing and building the filter, it is time to deploy it.
+
+#### Configuration
+
+WebAssembly support must be enabled in {{site.base_gateway}}:
+
+```console
+$ echo "wasm = on" >> /etc/kong.conf
+```
+
+Additionally, the `wasm_filters_path` parameter must be configured in order for
+{{site.base_gateway}} to load the filter at runtime. During local development,
+when a short feedback loop is desirable, you may want to set this parameter to
+your build toolchain's output directory (wherever the compiled
+`<filter-name>.wasm` file is produced):
+
+```console
+echo "wasm_filters_path = /path/to/my_filter/build" >> /etc/kong.conf
+```
+
+#### Link to a Kong Service and Route
+
+Now, the next step is to create a Kong filter chain entity using the
+filter created in the previous step and associate it with a Kong route
+or service. This will be done by creating a declarative yaml config file
+at `kong.yml`:
+
+```yaml
+---
+_format_version: '3.0'
+services:
+  - name: my-wasm-service
+    url: http://mockbin.org
+    routes:
+      - name: my-wasm-route
+        paths:
+          - /
+    filter_chains:
+      - name: my-wasm-demo
+        filters:
+          - name: my_filter
+            config: >-
+              {
+                "my_greeting": "Hello from WasmX!"
+              }
+```
+
+```console
+$ echo "database = off" >> /etc/kong.conf
+$ echo "declarative_config = $PWD/kong.yml" >> /etc/kong.conf
```

Comment:
Just a thought, with no strong feelings either way: I wonder if the tutorial should use `export` to environment variables instead, since a lot of people (most people? _\[citation needed\]_) run Kong on containers (most operators? do we know?) and then these are an easier transfer to those environments :thinking:

Reply:
Makes sense to me. I made the change.